presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -213,6 +217,10 @@ public static HiveSplitSource bucketed(
                     @Override
                     public <O> ListenableFuture<O> borrowBatchAsync(OptionalInt bucketNumber, int maxSize, Function<List<InternalHiveSplit>, BorrowResult<InternalHiveSplit, O>> function)
```

Comment:
It would be nice to add a comment here saying that the caller of this `borrowBatchAsync ` method waits until the returned future is done, and then calls this method again (as the `borrowBatchAsync` inside the `AsyncFunction` below may be confusing for the reader). You can merge that comment with the comment on L221 below.

Reply:
Moved comment above and noted "call this method again"